internal/resources/azure/log_analytics_workspace.go

```diff
@@ -0,0 +1,236 @@
+
+	skuCapacityReservation = "CapacityReservation"
+	skuPerGB2018           = "PerGB2018"
+	skuFree                = "Free"
+	skuFilterPAYG          = "Pay-as-you-go"
+
+	logRetentionFreeTierLimit = 30
+)
+
+var (
+	// unsupportedLegacySkus represents skus that Infracost doesn't support because these skus are
+	// legacy pricing tiers: https://docs.microsoft.com/en-us/azure/azure-monitor//logs/manage-cost-storage#legacy-pricing-tiers
+	unsupportedLegacySkus = map[string]struct{}{
+		"unlimited": {},
+		"standard":  {},
+		"premium":   {},
+		"pernode":   {},
+	}
+)
+
+// LogAnalyticsWorkspace struct represents an Azure Monitor log workspace. A workspace consolidates data
+// from multiple sources into a single data lake. A workspace defines:
+//
+//		1. The geographic location of the data.
+//		2. Access rights that define which users can access data.
+//		3. Configuration settings such as the pricing tier and data retention.
+//
+// Resource information: https://azure.microsoft.com/en-gb/services/monitor/
+// Pricing information: https://azure.microsoft.com/en-gb/pricing/details/monitor/
+type LogAnalyticsWorkspace struct {
+	Address string
+	Region  string
+	SKU     string
+
+	ReservationCapacityInGBPerDay int64
+	RetentionInDays               int64
+
+	MonthlyLogDataIngestionGB           *float64 `infracost_usage:"monthly_log_data_ingestion_gb"`
+	MonthlyAdditionalLogDataRetentionGB *float64 `infracost_usage:"monthly_additional_log_data_retention_gb"`
+	MonthlyLogDataExportGB              *float64 `infracost_usage:"monthly_log_data_export_gb"`
+}
+
+// LogAnalyticsWorkspaceUsageSchema defines a list which represents the usage schema of LogAnalyticsWorkspace.
+var LogAnalyticsWorkspaceUsageSchema = []*schema.UsageItem{
+	{
+		Key:          "monthly_log_data_ingestion_gb",
+		DefaultValue: 0,
+		ValueType:    schema.Float64,
+	},
+	{
+		Key:          "monthly_additional_log_data_retention_gb",
+		DefaultValue: 0,
+		ValueType:    schema.Float64,
+	},
+	{
+		Key:          "monthly_log_data_export_gb",
+		DefaultValue: 0,
+		ValueType:    schema.Float64,
+	},
+}
+
+// PopulateUsage parses the u schema.UsageData into the LogAnalyticsWorkspace.
+// It uses the `infracost_usage` struct tags to populate data into the LogAnalyticsWorkspace.
+func (r *LogAnalyticsWorkspace) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+// BuildResource builds a schema.Resource from a valid LogAnalyticsWorkspace struct.
+// The returned schema.Resource can have 3 potential schema.CostComponent associated with it:
+//
+//		1. Log data ingestion, which can be either:
+//			a) Pay-as-you-go, which is only valid for a sku of PerGB2018 and uses a usage param
+//			b) Billed per commitment tiers, which is only valid for a sku of CapacityReservation
+//		2. Log retention, which is free up to 31 days and then is billed per GB of data retained after the free limit.
+//		   Additional GB used comes from the usage param.
```

Comment:
The last sentence seems to be a bit ambiguous: I understood the docs as there is no "additional" data, all data the resource has is billed per _additional_ retention days. We may also rename `MonthlyAdditionalLogDataRetentionGB` to `MonthlyLogDataRetentionGB` to specify how much data users retain.

Or is it that users remove some log data after 31 days and keep the rest that will be billed?

What do you think?

> Data retained beyond these no-charge periods will be charged for each GB of data retained for a month (pro-rated daily).

Reply:
Yes you are correct the user is billed per additional day that data is stored past the 31 day mark. Users would have the option to remove some log data and keep some. So the billable unit is GB. e.g. a user could store 4GB of data from 1st of jan, clean the data on the 2nd of Jan to 2GB  - then this 2GB of data starts to be chargeable on the 1st of Feb. I'll change the wording to how you have suggested - I think keeping the `MonthlyLogDataRetentionGB` probably is less confusing (although it still is confusing!)

Reply:
fixed in https://github.com/infracost/infracost/pull/1367/commits/3262c570a559e70dd0f7736723b4022ddd4af91f